### ISSUE TITLE
Github Enterprise support

### ISSUE THREAD

**Issue by lai** [HubPress/hubpress.io#123]
I was able to change a few lines in the compiled source code to make it work on Github Enterprise before when the compiled code was not minified. Within hubpress the only changes necessary is to customize the base path for custom domain and repo directory. Instead of
```
https://username.github.io/hubpress.io or https://username.github.io
```
It will be
```
https://custom.domain.name/pages/username/hubpress.io or https://custom.domain.name/pages/username
```
This can be easily achieved in hubpress. However [`github.js`](https://github.com/michael/github) as a dependency doesn't have official support yet for Github Enterprise. The enterprise version has largely the same API endpoints as the public version. I was able to replace the base url within the library code and it seemed that everything just worked fine.

@anthonny please advice what would you like to do about this. If this is deemed as not fixable for the time being, we can close this out

**Comment by anthonny** [comment]
:+1: Thank you @lai!
I'll see to integrate these elements quickly

**Comment by jaredmorgs** [comment]
Is this possible with the API changes since this ticket was raised @anthonny ? If it is possible to do, perhaps add it to milestone 10?

**Comment by anthonny** [comment]
Yes i think it is still possible, i could add an entry in the settings panel of HubPress to say if the repository is an enterprise one.
Then in the github plugin i'll be able to add the /pages if necessary.
